Mlem/App/Globals/Definitions/Palette.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ColorProvider.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-06.
+//
+
+import Foundation
+import SwiftUI
+
+protocol PaletteProviding {
+    // basics
+    var primary: Color { get }
+    var background: Color { get }
+    var secondaryBackground: Color { get }
+    var tertiaryBackground: Color { get }
+    var accent: Color { get }
+    var uiAccent: UIColor { get }
```

Comment:
`UIColor` has an initialiser that accepts a `Color`, so we can avoid having a separate `uiAccent` property here.

```swift
UIColor(Color.red)
```